vhdbuilder/packer/packer_source.sh

```diff
@@ -124,6 +124,32 @@ copyPackerFiles() {
   PVT_HOST_SVC_DEST=/etc/systemd/system/reconcile-private-hosts.service
   cpAndMode $CSE_REDACT_SRC $CSE_REDACT_DEST 600
 
+  if grep -q "kata" <<< "$FEATURE_FLAGS"; then
+    # KataCC SPEC file assumes kata config points to the files exactly under this path
+    KATA_CONFIG_DIR=/var/cache/kata-containers/osbuilder-images/kernel-uvm/
+    KATACC_CONFIG_DIR=/opt/confidential-containers/share/kata-containers
+
+    IGVM_DEBUG_BIN_SRC=/home/packer/kata-containers-igvm-debug.img
+    IGVM_DEBUG_BIN_DEST=$KATACC_CONFIG_DIR/kata-containers-igvm-debug.img
+    cpAndMode $IGVM_DEBUG_BIN_SRC $IGVM_DEBUG_BIN_DEST 0755
+
+    IGVM_BIN_SRC=/home/packer/kata-containers-igvm.img
+    IGVM_BIN_DEST=$KATACC_CONFIG_DIR/kata-containers-igvm.img
+    cpAndMode $IGVM_BIN_SRC $IGVM_BIN_DEST 0755
+
+    KATA_INITRD_SRC=/home/packer/kata-containers-initrd-base.img
+    KATA_INITRD_DEST=$KATA_CONFIG_DIR/kata-containers-initrd.img
+    cpAndMode $KATA_INITRD_SRC $KATA_INITRD_DEST 0755
+
+    KATACC_IMAGE_SRC=/home/packer/kata-containers.img
+    KATACC_IMAGE_DEST=$KATACC_CONFIG_DIR/kata-containers.img
+    cpAndMode $KATACC_IMAGE_SRC $KATACC_IMAGE_DEST 0755
```

Comment:
755 is a very broad scope of access, other files have 600 or 644, any reason why we are doing 755?

Reply:
They needs to be readable and executable for the kata-cc image scenario. I have tested with 644 before that it seems 755 is required.